client/authority-discovery/src/error.rs

```diff
@@ -46,4 +46,8 @@ pub enum Error {
 	EncodingDecodingScale(codec::Error),
 	/// Failed to parse a libp2p multi address.
 	ParsingMultiaddress(libp2p::core::multiaddr::Error),
+	/// Failed to sign using a specific public key
+	MissingSignature(usize),
```

Comment:
I don't understand how the index would be useful for someone needing to debug this. Wouldn't something like the public key be more helpful?

Reply:
Done,,, the enum variant should receive `CryptoTypePublicPair`